torch/_inductor/ir.py

```diff
@@ -4680,6 +4681,44 @@ def codegen(self, wrapper):
             wrapper.writeline(f"{self.get_name()} = None")
 
 
+class RecordScalar(ExternKernel):
```

Comment:
Why can't this just go through the Inductor fallback mechanism? the custom op itself can, in its implementation, log the scalar to a file

Reply:
Yes we can, but I'm doing this to flesh out ideas of further extension, if we allow ```_record_scalar``` to call python functions to log sth to a file, but probably it should be part of the callback op in the future.

Reply:
Inductor will probably have special handling for the with_effects HOP (cc @angelayi) (with_effects is basically just the callback op). So yeah that will likely get its own Kernel here